src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/admin/ClustersController.kt

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.digital.hmpps.personrecord.api.controller.admin
+
+import io.swagger.v3.oas.annotations.Hidden
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.web.PageableDefault
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.personrecord.api.constants.Roles
+import uk.gov.justice.digital.hmpps.personrecord.api.model.admin.cluster.AdminCluster
+import uk.gov.justice.digital.hmpps.personrecord.api.model.admin.cluster.SourceSystemComposition
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonKeyRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.model.types.UUIDStatusType
+
+@RestController
+class ClustersController(
+  private val personKeyRepository: PersonKeyRepository,
+) {
+
+  @Hidden
+  @PreAuthorize("hasRole('${Roles.PERSON_RECORD_ADMIN_READ_ONLY}')")
+  @GetMapping("/admin/clusters")
+  suspend fun getClusters(
+    @PageableDefault(size = DEFAULT_PAGE_SIZE) pageable: Pageable,
+  ): Page<AdminCluster> {
+    val evaluatedPageable: Pageable = pageable.check()
+    return withContext(Dispatchers.IO) {
+      personKeyRepository.findAllByStatus(UUIDStatusType.NEEDS_ATTENTION, evaluatedPageable)
+    }.map {
+      AdminCluster(
+        uuid = it.personUUID.toString(),
+        recordComposition = SourceSystemComposition(
+          nomis = it.personEntities.getRecordCountBySourceSystem(SourceSystemType.NOMIS),
+          delius = it.personEntities.getRecordCountBySourceSystem(SourceSystemType.DELIUS),
+          commonPlatform = it.personEntities.getRecordCountBySourceSystem(SourceSystemType.COMMON_PLATFORM),
+          libra = it.personEntities.getRecordCountBySourceSystem(SourceSystemType.LIBRA),
```

Comment:
I know this was on the mockups but is it really useful? It's quite a lot of work to get it, as seen here.